packages/web/src/store/tipping/utils.ts

```diff
@@ -0,0 +1,187 @@
+import { ID } from 'common/models/Identifiers'
+import { UserTip } from 'common/models/Tipping'
+import { Nullable } from 'common/utils/typeUtils'
+import { FEED_TIP_DISMISSAL_TIME_LIMIT } from 'utils/constants'
+
+const RECENT_TIPS_KEY = 'recent-tips'
+
+type RecentTipsStorage = {
+  dismissed: boolean
+  minSlot: number
+  lastDismissalTimestamp: Nullable<number>
+}
+
+const getRecentTipsStorage = () => {
+  if (!window.localStorage) {
+    return null
+  }
+
+  const value = window.localStorage.getItem(RECENT_TIPS_KEY)
+  if (!value) {
+    return null
+  }
+
+  return JSON.parse(value) as RecentTipsStorage
+}
+
+export const getMinSlotForRecentTips = () => {
+  const storage = getRecentTipsStorage()
+  return storage ? storage.minSlot : null
+}
+
+export const getDismissedSlotForRecentTips = () => {
+  const storage = getRecentTipsStorage()
+  return storage ? storage.dismissed : null
+}
+
+export const checkRecentTip = ({
+  userId,
+  recentTips
+}: {
+  userId: ID
+  recentTips: UserTip[]
+}) => {
+  if (recentTips.length === 0) {
+    return null
+  }
+
+  /**
+   * The list only comprises of recent tips.
+   * Sort them tips by least recent to parse through oldest tips first.
+   */
+  const sortedTips = recentTips.sort((tip1, tip2) => tip1.slot - tip2.slot)
+
+  /** Return oldest of the recent tips if no local storage */
+  if (!window.localStorage) {
```

Comment:
i thinkk we are good, but just wanted to double check this won't run on native yeah?

Reply:
are you asking if the native mobile code can reach this flow? i think the answer is yes because the tipping sagas calls this function when fetching recent tips on the feed page

Reply:
oh yeah nvm this is okay since this code will run in the webview on native, nvm!!